src/caffe/util/hdf5.cpp

```diff
@@ -27,7 +27,34 @@ void hdf5_load_nd_dataset_helper(
   status = H5LTget_dataset_info(
       file_id, dataset_name_, dims.data(), &class_, NULL);
   CHECK_GE(status, 0) << "Failed to get dataset info for " << dataset_name_;
-  CHECK_EQ(class_, H5T_FLOAT) << "Expected float or double data";
+  switch (class_) {
+  case H5T_FLOAT:
+    LOG(INFO) << "Datatype class: H5T_FLOAT";
```

Comment:
Could you remove this and the below `LOG(INFO)`s (or change to `VLOG`, or one of the `LOG` variations that only happens the first time it's hit, if you prefer)?  I think this creates too much noise while training nets with an HDF5DataLayer, especially if using relatively small HDF5 files.